### PR TITLE
Upgrade Gatsby to v4 (and everything else)

### PR DESCRIPTION
**Pull request by kylegach**
- Update for all patch & minor package upgrades
- Update Gatsby to v4 
    - Switch from `gatsby-plugin-emotion` to `@emotion/babel-plugin`
    - Upgrade Storybook to webpack5
    - Remove patch-package
- Update other packages 
    - Except linting-related
- Update lint-related packages
    - Fix lint errors

### CODE REVIEW

**Comment by kylegach** [comment]
@winkerVSbecks — Any ideas on the [CircleCI build error](https://app.circleci.com/pipelines/github/chromaui/learnstorybook.com/1313/workflows/ca71ab5d-5ba9-4ebf-8c0a-ebc374a35845/jobs/1323)? That's inscrutable to me...

**Comment by winkerVSbecks** [comment]
@kylegach did it just run out of CPUs? Try adding `GATSBY_CPU_COUNT=1` to the build command in circle config

**Comment by kylegach** [comment]
> Try adding `GATSBY_CPU_COUNT=1` to the build command in circle config

That did the trick! 🎉

**Comment by kylegach** [comment]
@winkerVSbecks — This is ready for review!

Chromatic says UI Review is disabled (but I see it on other PRs... 🤷‍♂️) and the status checks aren't appearing, so here's the list of builds: https://www.chromatic.com/builds?appId=5cf841a3f3e3d200208ffc74&branch=new-site-design

File: .storybook/preview.js [hunk 1]
```diff
@@ -35,16 +33,13 @@ global.___loader = {
 global.__PATH_PREFIX__ = '';
 global.__BASE_PATH__ = '';
 
-// This is to utilized to override the window.___navigate method Gatsby defines and uses to report what path a Link would be taking us to if it wasn't inside a storybook
-window.___navigate = pathname => {
```
Comment (winkerVSbecks) [inline .storybook/preview.js:39]: You'll still need this

File: package.json [hunk 2]
```diff
@@ -4,33 +4,31 @@
...
-    "gatsby-plugin-layout": "^1.0.15",
-    "gatsby-plugin-react-helmet": "^3.0.12",
-    "gatsby-plugin-sharp": "^2.14.1",
-    "gatsby-plugin-sitemap": "^2.1.0",
+    "@storybook/design-system": "7.8.5",
+    "@storybook/theming": "^6.5.10",
+    "cheerio": "^1.0.0-rc.12",
+    "formik": "^2.2.9",
+    "gatsby": "^4.22.0",
+    "gatsby-plugin-facebook-pixel": "^1.0.8",
```
Comment (winkerVSbecks) [inline package.json:12]: This is a nice to have but, I don't think `gatsby-plugin-facebook-pixel` is used. We certainly don't track those metrics.

**Comment by winkerVSbecks** [comment]
@kylegach CleanShot 2022-09-08 at 12 41 59@2x

The chevron is missing in the language dropdown. I think the icon might have been renamed cause it's rendering the svg but no path

**Review by winkerVSbecks — APPROVED** [review]
Thanks for doing this big upgrade! Couple of minor suggestions and one bug. Good to merge after you fix that.

**Pull request merged**